services/QuillLMS/app/controllers/teachers/classrooms_controller.rb

```diff
@@ -173,7 +173,9 @@ def transfer_ownership
   end
 
   private def format_classrooms_for_index
-    classrooms = Classroom.unscoped.order(created_at: :desc).joins(:classrooms_teachers).where(classrooms_teachers: {user_id: current_user.id})
+    has_classroom_order = ClassroomsTeacher.where(user_id: current_user.id).all? { |classroom| classroom.order }
+    classrooms = Classroom.unscoped.joins(:classrooms_teachers).where(classrooms_teachers: {user_id: current_user.id})
+    classrooms = has_classroom_order ? classrooms.includes(:classrooms_teachers).order('classrooms_teachers.order') : classrooms.order(created_at: :desc)
```

Comment:
Might be worth adding a test case for this behavior, just to ensure that we don't break sorting later on somehow.  So a case where they all have order, and a case where only some have order, would probably cover us here.